File: Andriod/Constructor/construction/src/main/java/com/palitri/openiot/construction/framework/web/models/Project.java
package com.palitri.openiot.construction.framework.web.models;

import com.palitri.openiot.construction.framework.tools.utils.ArrayUtils;
import com.palitri.openiot.construction.framework.web.models.configurations.project.ProjectConfiguration;

import org.json.JSONException;
import org.json.JSONObject;

import java.io.Serializable;

public class Project implements Serializable {
    public String projectId;
    public String name;
    public String userId;
    public ProjectConfiguration boardConfig;
    public String schemeCode;
    public String scriptCode;
    public String creationDate;

    public Project()
    {
        this.boardConfig = new ProjectConfiguration();
    }

    public boolean LoadFromJSON(JSONObject jsonObject)
    {
        try {
            this.projectId = jsonObject.has("projectId") ? jsonObject.getString("projectId") : null;
            this.name = jsonObject.has("name") ? jsonObject.getString("name") : null;
            this.userId = jsonObject.has("userId") ? jsonObject.getString("userId") : null;
            this.schemeCode = jsonObject.has("schemeCode") ? jsonObject.getString("schemeCode") : null;
            this.scriptCode = jsonObject.has("scriptCode") ? jsonObject.getString("scriptCode") : null;
            this.creationDate = jsonObject.has("creationDate") ? jsonObject.getString("creationDate") : null;

            this.boardConfig = new ProjectConfiguration();
            if (jsonObject.has("boardConfig"))
                this.boardConfig.LoadFromJSON(jsonObject.getJSONObject("boardConfig"));
        } catch (JSONException e) {
            e.printStackTrace();
            return false;
        }

        return true;
    }

    public byte[] getCompiledSchemeCode()
    {
        return ArrayUtils.commaSeparatedValuesToBytes(this.schemeCode);
    }

    public byte[] getCompiledScriptCode()
    {
        return ArrayUtils.commaSeparatedValuesToBytes(this.scriptCode);
    }
}

File: Windows/OpenIoT.App/OpenIoT.Lib/Web/Models/Project.cs
using OpenIoT.Lib.Tools.Utils;
using OpenIoT.Lib.Web.Models.Configurations.Project;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OpenIoT.Lib.Web.Models
{
    public class Project
    {
        public string ProjectId { get; set; }
        public string Name { get; set; }
        public string UserId { get; set; }
        public ProjectConfiguration BoardConfig { get; set; }
        public string SchemeCode { get; set; }
        public string ScriptCode { get; set; }
        public DateTime CreationDate { get; set; }
        public DateTime UpdatedDate { get; set; }


        public byte[] GetCompiledSchemeCode()
        {
            return ArrayUtils.CommaSeparatedValuesToBytes(this.SchemeCode);
        }

        public byte[] GetCompiledScriptCode()
        {
            return ArrayUtils.CommaSeparatedValuesToBytes(this.ScriptCode);
        }
    }
}
